synapse/api/auth.py

```diff
@@ -205,31 +202,39 @@ async def get_user_by_req(
                         device_id="dummy-device",  # stubbed
                     )
 
-                return synapse.types.create_requester(user_id, app_service=app_service)
+                requester = synapse.types.create_requester(
+                    user_id, app_service=app_service
+                )
+
+                request.requester = user_id
```

Comment:
This implies some potential clean-up if each request has a requester on it... (I think we frequently pass the `SynapseRequest` and `Requester` into handlers). Nothing to really do here though. Just noting.